poutyne/framework/callbacks/wandb_logger.py

```diff
@@ -0,0 +1,256 @@
+        """
+        dummies_batch = torch.randn(self.training_batch_shape)
+        save_path = self.run.dir + "/" + self.run.name + "_model.onnx"
+        torch.onnx.export(self.model.network, dummies_batch, save_path)
+        self.run.save(save_path)
+
+    def on_train_begin(self, logs: Dict):
+        super().on_train_begin(logs)
+        if self.log_gradient_frequency is not None:
+            self._watch_gradient()
+
+        if self.training_batch_shape is not None:
+            self._save_architecture()
+
+    def log_config_params(self, config_params: Dict) -> None:
+        """
+        Args:
+            config_params Dict:
+                Dictionary of config parameters of the training to log, such as number of epoch, loss function,
+                optimizer etc.
+        """
+        self.run.config.update(config_params)
+
+    def _on_train_batch_end_write(self, batch_number: int, logs: Dict) -> None:
+        """
+        Log the batch metric.
+        """
+        if self.batch_granularity:
+            train_metrics = {key: value for (key, value) in logs.items() if "val_" not in key}
+            train_metrics = {"training": {"batch": train_metrics}}
+            self._log_metrics(train_metrics)
+
+    def _on_epoch_end_write(self, epoch_number: int, logs: Dict) -> None:
+        """
+        Log the epoch metric.
+        """
+        train_metrics = {key: value for (key, value) in logs.items() if "val_" not in key}
+        val_metrics = {key.replace("val_", ""): value for (key, value) in logs.items() if "val_" in key}
+        learning_rate = self._get_current_learning_rates()
+
+        if self.batch_granularity:
+            train_metrics = {"training": {"epoch": train_metrics}}
+        else:
+            train_metrics = {"training": train_metrics}
+
+        val_metrics = {"validation": val_metrics}
+
+        self._log_metrics(train_metrics, step=epoch_number)
+        self._log_metrics(val_metrics, step=epoch_number)
+        self._log_params(learning_rate, step=epoch_number)
+
+    def _on_train_end_write(self, logs: Dict):
+        if self.checkpoints_path is not None:
+            self._log_artifacts([self.checkpoints_path], "Checkpoints", artifact_type="Model-weights")
+
+    def _log_metrics(self, metrics: Dict, step: int = None):
+        """
+        Log metrics for a specific step.
+        Args:
+            metrics (Dict): the metrics to log in the form of a dictionary.
+            step (int): the corresponding step.
+        """
+        if self.batch_granularity:
+            self.run.log(metrics)
+        else:
+            self.run.log(metrics, step=step)
+
+    def _log_params(self, params: Dict, step: int):
+        """
+        Log parameters for a specific step.
+        This functions logs parameters as metrics since wandb doesn't support
+        parameter logging. However, the logged parameters are prepended by the keyword
+        `parameter` so as to easily identify them.
```

Comment:
```suggestion
Log parameters for a specific step.
        This functions logs parameters as metrics since wandb doesn't support
        parameter logging. However, the logged parameters are appended by the keyword
        `parameter` to easily identify them.
```